refinery/ui/source/js/commons/ctrls/permission-editor-ctrl.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Permission Editor Ctrl
+ * @namespace PermissionEditorCtrl
+ * @desc Main controller for the permission modal component.
+ * @memberOf refineryApp
+ */
+
+(function () {
+  'use strict';
+
+  angular
+    .module('refineryApp')
+    .controller('PermissionEditorCtrl', PermissionEditorCtrl);
+
+  PermissionEditorCtrl.$inject = [
+    '$log',
+    '_',
+    'sharingService',
+    'permissionService'
+  ];
+
+  function PermissionEditorCtrl (
+    $log,
+    _,
+    sharingService,
+    permissionService
+  ) {
+    var vm = this;
+    vm.cancel = cancel;
+    vm.permissions = permissionService.permissions;
+    // Used as a shorthand to avoid complicated permission checking in `ngRepeat`
+    vm.permissionLevel = {
+      none: {
+        read: false,
+        read_meta: false,
+        change: false
+      },
+      read_meta: {
+        read: false,
+        read_meta: true,
+        change: false
+      },
+      read: {
+        read: true,
+        read_meta: true,
+        change: false
+      },
+      edit: {
+        read: true,
+        read_meta: true,
+        change: true
+      }
+    };
+    vm.save = save;
+
+    /**
+     * Cancel permission editing.
+     * @type  {function}
+     */
+    function cancel () {
+      permissionService.getPermissions(vm.resolve.config.uuid);
```

Comment:
This might be a naming issue, or I might be confused: I would expect `getPermissions` to return a value, and not to have any side-effects, but that can't be the way it works here?